ui_test/src/lib.rs

```diff
@@ -293,50 +316,111 @@ fn check_test_result(
         comments,
     );
     // Check error annotations in the source against output
-    check_annotations(&stderr, errors, config, revision, comments);
+    check_annotations(diagnostics, path, errors, config, revision, comments);
 }
 
 fn check_annotations(
-    unnormalized_stderr: &str,
+    mut diagnostics: Diagnostics,
+    path: &Path,
     errors: &mut Errors,
     config: &Config,
     revision: &str,
     comments: &Comments,
 ) {
-    let mut found_annotation = false;
+    let mut highest_annotation_level = Level::Error;
```

Comment:
This is only used in the second block I think. Can you move down its declaration?